MyAddin/Form1.cs

```diff
@@ -0,0 +1,317 @@
+﻿using System;
+using System.IO;
+using System.Windows.Forms;
+using System.Xml;
+
+namespace MyAddin
+{
+    public partial class Form1 : Form
+    {
+        private XmlDocument docXML = new XmlDocument();
+
+        //this is the path where xml files are saved and loaded
+        //the input and output file will be found in AppData/roaming folder
+        string path = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
+        string fileName = Path.Combine(Environment.GetFolderPath(
+                Environment.SpecialFolder.ApplicationData), "input.xml");
+
+        string fileName1 = Path.Combine(Environment.GetFolderPath(
+                Environment.SpecialFolder.ApplicationData), "GenericFTS.xml");
+        String name1 = "";
+
+        // Xml tag for node, e.g. 'node' in case of <node></node>
+        public const string XmlNodeTag = "node";
+
+        // Xml attributes for node e.g. <node text="Sporadic Timing Failure"></node>
+        public const string XmlNodeTextAtt = "text";
+
+        public Form1()
+        {
+            try
+            {
+                InitializeComponent();
+                //this will chk if xml exist and load that
+                if (File.Exists(fileName))
+                {
+                    LoadXml(); //if xml exist then it will load the xml
+                    treeView1.ExpandAll();
+                }
+                else
+                {
+                    GenericFTS(); //else generic FTS loaded
+                    LoadXml();
+                    treeView1.ExpandAll();
+                }
+            }
+            catch (Exception e)
+            {
+                MessageBox.Show("Invalid XML", "Error", MessageBoxButtons.OK);
+                Console.WriteLine(e);
+            }
+        }
+
+        public void GenericFTS()
+        {
+            try
+            {
+                treeView1.Nodes.Add("Failure");
+                treeView1.Nodes[0].Nodes.Add("Sporadic Timing Failure");
+                treeView1.Nodes[0].Nodes[0].Nodes.Add("Sporadic Duration Failure");
+                treeView1.Nodes[0].Nodes[0].Nodes[0].Nodes.Add("SporadicProlongation");
+                treeView1.Nodes[0].Nodes[0].Nodes[0].Nodes[0].Nodes.Add("Permanent Prolongation");
+                treeView1.Nodes[0].Nodes[0].Nodes[0].Nodes.Add("Permanent Duration Failure");
+                treeView1.Nodes[0].Nodes[0].Nodes[0].Nodes[1].Nodes.Add("Permanent Prolongation");
+                treeView1.Nodes[0].Nodes[0].Nodes[0].Nodes[1].Nodes.Add("Permament Interupt");
+            }
+            catch (Exception)
+            {
+                MessageBox.Show("Not Valid GenericFTS", "Error", MessageBoxButtons.OK);
+            }
+
+        }
+
+        public void LoadXml()
+        {
+            try
+            {
+                ReadXMLFile(treeView1, fileName);
+            }
+            catch (Exception)
```

Comment:
make use of exception block and throw exception error message atleast on console.